gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -508,6 +526,26 @@ public static BlobSourceOption metagenerationMatch(long metageneration) {
     public static BlobSourceOption metagenerationNotMatch(long metageneration) {
       return new BlobSourceOption(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH, metageneration);
     }
+
+    static BlobSourceOption[] setGeneration(BlobId blobId, Iterable<BlobSourceOption> options) {
```

Comment:
You are right, sorry. I will look into this tomorrow morning, it's probably as simple as adding a method `optionMap(blobId, options)`.  Sorry:)